File: backend/Margorak.Api/Dto/CombatantHabitatDto.cs
using Margorak.Api.Models;

namespace Margorak.Api.Dto
{
    public class CombatantHabitatDto
    {
        public int CombatantHabitatId { get; set; }
        public int TerrainTypeId { get; set; } 
        public int LocX1 {  get; set; }
        public int LocY1 { get; set; }
        public int LocX2 {  get; set; }
        public int LocY2 { get; set; }
        public int Probability {  get; set; }
        public int AmbushProbability {  get; set; }

    }
}

File: frontend/src/app/shared/dto/combatant-habitat.dto.ts
export interface CombatantHabitatDto {
  combatantId: number;
  terrainTypeId: number;
  locX1: number;
  locY1: number;
  locX2: number;
  locY2: number;
  probability: number;
  ambushProbability: number;
}

export interface HabitatTerrainTypeDto {
  id: number;
  name: string;
}
